### DISCUSSION TITLE
why is my custom whatsapp driver not working on conversation but if I'm not inside conversation it works

### DISCUSSION THREAD

**Discussion by zerexei** [botman/botman#1383] [Q&A] [answered]
### my driver
> https://github.com/zerexei/botman-driver-whatsapp/blob/main/WhatsAppDriver.php

### controller
```php
$botman->hears(".*", function (BotMan $bot) {
  $bot->reply("Hello yourself"); // this works
  
  $recipientId = $userId = $request->entry[0]['changes'][0]['value']['messages'][0]['from'] ?? '';

  // no response but functions are called
  $bot->startConversation(new FacebookConversation, $recipientId, WhatsAppDriver::class);
});

$botman->listen();
```

### conversation
```php
use BotMan\BotMan\Messages\Conversations\Conversation;

class FacebookConversation extends Conversation
{
    public function run()
    {
       $this->say("Hello yourself");
    }
}
```

**Comment by filippotoso** [comment]
The entry point method of a Conversation object is a public method named `run()`, not a protected method named `start()`:

https://botman.io/2.0/conversations

https://github.com/botman/botman/blob/2.0/src/Messages/Conversations/Conversation.php#L242

**Reply by zerexei** [reply]
my bad, im extending it to my BotConversation class which has the run() method that calls the start method

**Reply by filippotoso** [reply]
Botman with any other driver works fine. The problem must be in your own code.

**Reply by zerexei** [reply]
just to be clear, is my botman driver setup correct? https://github.com/zerexei/botman-driver-whatsapp/blob/main/WhatsAppDriver.php

**Reply by zerexei** [reply]
here's the log, when botman enters `getConversationAnswer` method the **IncomingMessage** is empty I also tried overriding it but same result, it's not going to sendPayload

![image](https://github.com/botman/botman/assets/69749146/f6adad47-5c1d-4512-a682-5d70a5f14551)

**Comment by zerexei** [comment]
the driver is null inside conversation but outside it exists, tried adding it on AppServiceProvider but same result

![image](https://github.com/botman/botman/assets/69749146/3e95d8da-8169-431c-9dd4-a63a0d481a89)

**Comment by zerexei** [comment, marked as answer]
DRIVER_NAME needs to **match w/ filename w/out Driver**

const DRIVER_NAME = 'WhatsApp';
